lib/command_definitions.ts

```diff
@@ -2,8 +2,92 @@ import {CommandDefinition} from './command_definition';
 
 export let getNetworkConnection =
     new CommandDefinition<number>('getNetworkConnection', [], 'GET', '/network_connection');
-export let setNetworkConnection =
-    new CommandDefinition<void>('setNetworkConnection', ['type'], 'POST', '/network_connection');
+    'setNetworkConnection', ['type'], 'POST', '/network_connection',
+    (args: Array<number|boolean>) => {
+      if (typeof args[0] == 'boolean') {
+        // Transform into bitmask
+        return [(args[0] ? 1 : 0) + (args[1] ? 2 : 0) + (args[2] ? 4 : 0)];
+      } else {
+        return args;
+      }
+    });
+export let toggleAirplaneMode = new CommandDefinition<void>(
+    'toggleAirplaneMode', [], 'POST', 'appium/device/toggle_airplane_mode');
+export let toggleWiFi =
+    new CommandDefinition<void>('toggleWiFi', [], 'POST', 'appium/device/toggle_wifi');
+export let toggleData =
+    new CommandDefinition<void>('toggleData', [], 'POST', 'appium/device/toggle_data');
+
+export let toggleLocationServices = new CommandDefinition<void>(
+    'toggleLocationServices', [], 'POST', 'appium/device/toggle_location_services');
+export let getGeolocation =
+    new CommandDefinition<{latitude: number, longitude: number, altitude: number}>(
+        'getGeolocation', [], 'GET', '/location');
+export let setGeolocation = new CommandDefinition<void>(
+    'setGeolocation', ['location'], 'POST', '/location', (args: [number, number, number]) => {
+      return [{latitude: args[0] || 0, longitude: args[1] || 0, altitude: args[2] || 0}];
+    });
+
+export let getCurrentDeviceActivity = new CommandDefinition<string>(
+    'getCurrentDeviceActivity', [], 'GET', '/appium/device/current_activity');
+export let startDeviceActivity = new CommandDefinition<void>(
+    'startDeviceActivity', ['appPackage', 'appActivity', 'appWaitPackage', 'appWaitActivity'],
+    'POST', '/appium/device/start_activity', (args: string[]) => {
+      if (args.length == 2) {
+        args.length = 4;
```

Comment:
Oh, I'll just do the same thing as in the other PR.  Sorry, didn't check over this again in between